src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js

```diff
@@ -81,35 +90,52 @@ class SearchGroupFactory {
-            o.name = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.NAME) ||
-                trackedEntityAttribute.displayName;
-            o.shortName = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.SHORT_NAME) ||
-                trackedEntityAttribute.displayShortName;
-            o.formName = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.NAME) ||
-                trackedEntityAttribute.displayName;
-            o.description = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.DESCRIPTION) ||
-                trackedEntityAttribute.description;
+            const {
+                id,
+                translations,
+                displayName,
+                displayShortName,
+                description,
+                unique,
+                valueType,
+            } = searchAttribute.trackedEntityAttribute;
+
+            o.id = id;
+            o.name =
+              this._getAttributeTranslation(translations, translationPropertyNames.NAME)
+              || displayName;
+
+            o.shortName =
+              this._getAttributeTranslation(translations, translationPropertyNames.SHORT_NAME)
+              || displayShortName;
+
+            o.formName =
+              this._getAttributeTranslation(translations, translationPropertyNames.NAME)
+              || displayName;
+
+            o.description =
+              this._getAttributeTranslation(translations, translationPropertyNames.DESCRIPTION)
+              || description;
+
             o.displayInForms = true;
             o.displayInReports = searchAttribute.displayInList;
-            o.compulsory = !!trackedEntityAttribute.unique;
+            o.compulsory = !!unique;
             o.disabled = false;
-            o.type = SearchGroupFactory._getSearchAttributeValueType(trackedEntityAttribute.valueType, trackedEntityAttribute.unique);
+            o.type = SearchGroupFactory._getSearchAttributeValueType(valueType, unique);
         });
 
-        /* if (attribute.optionSet && attribute.optionSet.id ) {
-            element.optionSet = await buildOptionSet(
-                attribute.optionSet.id,
+        const { optionSetValue, optionSet } = searchAttribute.trackedEntityAttribute;
+
+        if (optionSetValue && optionSet.id) {
+            element.optionSet = await this.optionSetFactory.build(
                 element,
-                programAttribute.renderOptionsAsRadio,
-                programAttribute.renderType && programAttribute.renderType.DESKTOP && programAttribute.renderType.DESKTOP.type);
-        } */
+                optionSet.id,
+                searchAttribute.renderOptionsAsRadio,
+                null,
+                value => value,
+            );
+        }
 
         await Promise.resolve();
```

Comment:
Let's remove this old unnecessary code while we're at it.
```suggestion
```

Reply:
Good point